tests/unittests/test_dispatcher.py

```diff
@@ -457,20 +443,18 @@ async def _check_if_async_function_is_ok(self, host) -> Tuple[str, str]:
         return func_id, invoke_id, function_name
 
 
+@unittest.skipIf(sys.version_info.minor != 8,
+                 "Run the tests only for Python 3.8. In other platforms, "
+                 "as the default passed is None, the cpu_count determines the "
+                 "number of max_workers and we cannot mock the os.cpu_count() "
+                 "in the concurrent.futures.ThreadPoolExecutor")
 class TestThreadPoolSettingsPython38(TestThreadPoolSettingsPython37):
-    def setUp(self):
-        super(TestThreadPoolSettingsPython38, self).setUp()
-        self.mock_version_info = patch(
-            'azure_functions_worker.dispatcher.sys.version_info',
-            SysVersionInfo(3, 8, 0, 'final', 0))
-        self._over_max_workers: int = 10000
+    def setUp(self, version=SysVersionInfo(3, 8, 0, 'final', 0)):
+        super(TestThreadPoolSettingsPython38, self).setUp(version)
         self._allowed_max_workers: int = self._over_max_workers
-        self.mock_version_info.start()
 
     def tearDown(self):
-        os.environ.clear()
-        os.environ.update(self._pre_env)
-        self.mock_version_info.stop()
+        super(TestThreadPoolSettingsPython38, self).tearDown()
```

Comment:
If `self._over_max_workers: int = 10000` will not be used for 37, let's move it to 38.

Reply:
Done